### ISSUE TITLE
failure in ud/test_uct_pending.send_ooo_with_pending

### ISSUE THREAD

**Issue by yosefe** [openucx/ucx#1384]
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3122/label=hpc-test-node2,worker=3/console
```
16:41:30 [ RUN      ] ud/test_uct_pending.send_ooo_with_pending/1
16:41:30 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node2/worker/3/contrib/../test/gtest/uct/test_pending.cc:248: Failure
16:41:30 Value of: 0xdeadbeef + counter - 1
16:41:30   Actual: 3735928561
16:41:30 Expected: send_data
16:41:30 Which is: 3735928562
16:41:30 
[  FAILED  ] ud/test_uct_pending.send_ooo_with_pending/1, where GetParam() = ud/mlx5_0:1 (134 ms)
16:41:30 [----------] 1 test from ud/test_uct_pending (134 ms total)
...
16:41:48 
make: *** [test] Error 1
```

**Comment by yosefe** [comment]
http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3254/label=hpc-test-node2,worker=1/consoleFull
```
14:39:31 [ RUN      ] ud_mlx5/test_uct_pending.send_ooo_with_pending/1
14:39:31 /scrap/jenkins/scrap/workspace/hpc-ucx-pr-4/label/hpc-test-node2/worker/1/contrib/../test/gtest/uct/test_pending.cc:248: Failure
14:39:31 Value of: 0xdeadbeef + counter - 1
14:39:31   Actual: 3735928561
14:39:31 Expected: send_data
14:39:31 Which is: 3735928562
14:39:31 
[  FAILED  ] ud_mlx5/test_uct_pending.send_ooo_with_pending/1, where GetParam() = ud_mlx5/mlx5_0:1 (189 ms)
14:39:31 [----------] 1 test from ud_mlx5/test_uct_pending (189 ms total)
```

**Comment by yosefe** [comment]
fixed by #1424
